### PR TITLE
Thread save key detect

### PR DESCRIPTION
**Pull request by daschuer**
This fixes that a global buffers is used from the key detector object. Now every object used is own instance. 

This PR also removes some intermediate buffers.

Currently we don't even need to copy the buffer locally if we store the  but I have kept using it, because I do not want to alter the numbers from the original profiles which are taken from an academic paper.

### CODE REVIEW

**Comment by uklotzde** [comment]
I still don't spot any thread-safety issues?? Those pseudo-constant static arrays should not cause any harm, neither before nor after your changes. The elements will be initialized before the first analysis starts and are never changed ever after.

**Comment by uklotzde** [comment]
Oh, no, now I see it. It's in MathUtilities::circShift()!! OMG, that's really, really, bad and dangerous code.

**Comment by uklotzde** [comment]
One more reason to stick with Rust! The compiler prevents most of the ridiculous bullshit a mediocre developers (like me) try to

**Comment by Be-ing** [comment]
Does this mean #2009 can be reverted?

**Comment by uklotzde** [comment]
Not sure. The bug is not able to trigger a segfault and that's what I remember what we have experienced casually. This fix only ensures that no bogus values are returned.

**Comment by uklotzde** [comment]
There might be more and even more serious bugs hidden in the code, which looks like it has been ported literally from Mathlab without considering those aspects.

File: lib/qm-dsp/dsp/keydetection/GetKeyMode.cpp [hunk 1]
```diff
@@ -103,8 +102,19 @@ GetKeyMode::GetKeyMode( int sampleRate, float tuningFrequency,
     
     m_MajCorr = new double[m_BPO];
     m_MinCorr = new double[m_BPO];
-    m_Keys  = new double[2*m_BPO];
     
+    m_MajProfileNorm = new double[m_BPO];
```
Comment (uklotzde) [inline lib/qm-dsp/dsp/keydetection/GetKeyMode.cpp]: m_BPO is just the constant 36. Static allocation would work for all those arrays as well as a single dynamic allocation with partitioning/slicing.

**Comment by Be-ing** [comment]
#2009 was fixing a [bug](https://bugs.launchpad.net/mixxx/+bug/1813413) with incorrect analysis results, not a crash. So I think it's worth a try to revert that now.

**Comment by uklotzde** [comment]
Thanks @Be-ing for checking this out. That's what I planned to do now.

So this will finally allow us to unleash the full power of the multi-threaded analysis!!

**Comment by uklotzde** [comment]
Note: After this PR has been merged we can apply a revert commit from #2009. But without undoing the hotfix in #2010! I will take care of this later.

**Comment by uklotzde** [comment]
Please rebase after #2112 has now been merged.

**Comment by daschuer** [comment]
Done.

**Comment by daschuer** [comment]
This was superseded by https://github.com/mixxxdj/mixxx/pull/2136 and can be closed.

**Pull request closed without merging**